src/MongoDB/Command.c

```diff
@@ -69,6 +72,19 @@ static bool php_phongo_command_init(php_phongo_command_t *intern, zval *filter,
 		return false;
 	}
 
+	if (bson_iter_init (&iter, intern->bson) && bson_iter_find_descendant (&iter, "cursor.batchSize", &sub_iter) && BSON_ITER_HOLDS_INT(&sub_iter)) {
+		int64_t batch_size = bson_iter_as_int64(&sub_iter);
+		if (batch_size < 0) {
+			phongo_throw_exception(PHONGO_ERROR_INVALID_ARGUMENT TSRMLS_CC, "Expected \"batchSize\" option to be >= 0, %" PRId64 " given", batch_size);
+		}
+
+		if (batch_size > UINT32_MAX) {
+			phongo_throw_exception(PHONGO_ERROR_INVALID_ARGUMENT TSRMLS_CC, "Expected \"batchSize\" option to be <= %" PRIu32 ", %" PRId64 " given", UINT32_MAX, batch_size);
+		}
```

Comment:
IIRC, we discussed ignoring out-of-range values instead of throwing an exception, since this is internal behavior hidden from the user. Also, if they happen to use an invalid value, chances are the main command itself will yield an error. I'd suggest:

```c
if (batch_size >= 0 && batch_size <= UINT32_MAX) {
	intern->batch_size = (uint32_t) batch_size;
}
```

Note that if we were going to throw an exception here, it'd be important to return `false` after doing so. This all comes after the `if (EG(exception)) return false;` that trails BSON conversion, and it would therefore be possible for this `php_phongo_command_init()` function to return `true` so long as `php_phongo_command_init_max_await_time_ms()` didn't fail below.

Returning false on the first error will ensure that the calling function can act accordingly (technically not a problem in this case, since this function is the last statement of the Command constructor) and avoid cases where we end up throwing multiple exceptions (in that case, previous exceptions will get masked under the most recent and can only be accessed with [`Exception::getPrevious()`](http://php.net/manual/en/exception.getprevious.php)).